components/common/src/main/java/com/hotels/styx/NettyExecutor.java

```diff
@@ -70,7 +72,11 @@ private NettyExecutor(EventLoopGroup eventLoopGroup,
     }
 
     public void shut() {
-        eventLoopGroup.shutdownGracefully();
+        try {
+            eventLoopGroup.shutdownGracefully(0, 0, TimeUnit.SECONDS).await(5000);
```

Comment:
Should we wait here?

Reply:
Hi. It does wait for `shutdownGracefully` to come back.

Or do you refer to the `quietPeriod` setting? But we never really supported a graceful shutdown properly. Yet. I think we need to implement that on a "styx level".

The reason for immediate quiet and grace period is to avoid excessive waiting in functional test suite.

However I do think, going forward, we need to add timeout controls and a shutdown future to this API. Perhaps that should be a next increment, providing that the work is completed before the Styx 1.1 release. WDYT?

Reply:
The executors will only ever be closed when styx shuts. Therefore any shutdown future, etc doesn't really matter at the moment as the JVM is going to shut anyway. I would revisit this code if or once we'll allow consumers to shut injected executors.